common/persistence/dataInterfaces.go

```diff
@@ -389,6 +390,13 @@ type (
 		RecordVisibility    bool
 	}
 
+	// RecordWorkflowStartedTask identifites a transfer task for writing visibility open execution record
+	RecordWorkflowStartedTask struct {
```

Comment:
Why RecordWorkflowStartedTask is not persisted? In that case we will lose task right?

Reply:
I don't quite understand. The task is created as a transfer task and I think all transfer tasks will be written to persistence, and then the task will be loaded and processed by the transferQueue. Am I missing something here?

Reply:
I am also not sure I follow it looks like the correct thing is being done in historyEngine.go